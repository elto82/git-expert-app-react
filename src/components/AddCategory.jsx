import { useState } from 'react';

export const AddCategory = ({ onNewCategory }) => {

  const [ inputValue, setInputValue ] = useState('');

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
    //console.log(inputValue);
  };

  const onSutmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length <= 1) return;
    // setCategories(cat => [ inputValue, ...cat ]);
    onNewCategory(inputValue.trim());
    setInputValue("");
  };

  return (
    <form onSubmit={onSutmit}>
      <input
        type="text"
        placeholder="Buscar Gifs"
        value={inputValue}
        onChange={onInputChange}
      />

    </form>
  );
};
