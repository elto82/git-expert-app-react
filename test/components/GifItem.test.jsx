import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Test component <GifItem/>', () => {
  const title = 'giphy';
  const url = 'https://media.giphy.com/media/3o7TKXwuJQwzqQMf2p/giphy.gif';

  test('should match with snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('should show the title and url image', () => {
    const { container, getByText } = render(<GifItem title={title} url={url} />);
    expect(getByText(title)).toBeTruthy();
    expect(container.querySelector('img').src).toBe(url);
    const { src } = screen.getByRole('img');
    expect(src).toBe(url);
  });

  test('should show the title in the component', () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });

});