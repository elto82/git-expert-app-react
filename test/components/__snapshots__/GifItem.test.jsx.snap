// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component <GifItem/> should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="giphy"
      src="https://media.giphy.com/media/3o7TKXwuJQwzqQMf2p/giphy.gif"
    />
    <p>
      giphy
    </p>
  </div>
</div>
`;
